### PR TITLE
groups: update roles async and load RoleNeed

### PR DESCRIPTION
**Pull request by ntarocco**
* update groups async, to improve performance on login
* connect to the identity_loaded signal to load User/Role needs on login

### CODE REVIEW

File: invenio_oauthclient/contrib/keycloak/helpers.py [hunk 1]
```diff
@@ -85,12 +85,14 @@ def _get_user_info_from_endpoint(remote, config_prefix):
...
 
 
-def get_user_info(remote, resp_token):
+def get_user_info(remote, resp_token, from_token_only=False):
     """Get the user information from Keycloak.
 
     :param remote: The OAuthClient remote app
     :param resp_token: The response from the 'token' endpoint; expected to be a dict
         and to contain a JWT 'id_token'
+    :param from_token_only: return info o`nly from the token, without calling the user info
```
Comment (jrcastro2) [inline invenio_oauthclient/contrib/keycloak/helpers.py:94]:
```suggestion
    :param from_token_only: return info only from the token, without calling the user info
```

File: tests/test_handlers_ui.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 from flask import session, url_for
 from flask_login import current_user
 from flask_oauthlib.client import OAuth as FlaskOAuth
+from flask_principal import RoleNeed
```
Comment (jrcastro2) [inline tests/test_handlers_ui.py:15]: Looks like a leftover?

**Review by jrcastro2 — APPROVED** [review]
LGTM! :rocket:

**Comment by ntarocco** [comment]
Skipping tests as suggested in: https://github.com/inveniosoftware/invenio-app-rdm/issues/2258

**Pull request merged**